tests/tests/test_update_control.py

```diff
@@ -127,3 +127,67 @@ def test_update_control(
         )
 
         deploy.check_expected_status("finished", deployment_id)
+        deploy.check_expected_statistics(deployment_id, "success", 1)
+
+    def test_update_control_with_broken_map(
+        self, enterprise_no_client, valid_image,
+    ):
+        """
+        Schedule an update with an invalid map, which should fail.
+        """
+
+        u = User("", "bugs.bunny@acme.org", "whatsupdoc")
+        cli = CliTenantadm(containers_namespace=enterprise_no_client.name)
+        tid = cli.create_org("enterprise-tenant", u.name, u.pwd, "enterprise")
+        tenant = cli.get_tenant(tid)
+        tenant = json.loads(tenant)
+        ttoken = tenant["tenant_token"]
+
+        auth = Authentication(name="enterprise-tenant", username=u.name, password=u.pwd)
+        auth.create_org = False
+        auth.reset_auth_token()
+        devauth = DeviceAuthV2(auth)
+
+        enterprise_no_client.new_tenant_client("control-map-test-container", ttoken)
+        device = MenderDevice(enterprise_no_client.get_mender_clients()[0])
+        devauth.accept_devices(1)
+
+        deploy = Deployments(auth, devauth)
+
+        devices = list(
+            set([device["id"] for device in devauth.get_devices_status("accepted")])
+        )
+        assert 1 == len(devices)
+
+        wait_for_connect(auth, devices[0])
```

Comment:
Out of curiosity, why do you need to wait for connect?

Reply:
I don't know, actually. All this boiler plate code was copied from the previous test, which @oleorhagen wrote. I wouldn't be surprised if this was again copied from somewhere else that actually needs it. I'll merge this as is, but I can try removing them in a followup.

Reply:
https://github.com/mendersoftware/integration/pull/1460